edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/DeviceScopeController.cs

```diff
@@ -248,20 +265,22 @@ async Task<bool> AuthorizeActorAsync(IDeviceScopeIdentitiesCache identitiesCache
             return true;
         }
 
-        async Task SendResponse(EdgeHubScopeResult result)
+        async Task SendSuccessResponse(EdgeHubScopeResult result)
         {
-            this.Response.StatusCode = (int)result.Status;
+            this.Response.StatusCode = (int)HttpStatusCode.OK;
+            var resultJsonContent = JsonConvert.SerializeObject(result);
+            var resultUtf8Bytes = Encoding.UTF8.GetBytes(resultJsonContent);
 
-            if (result.Status == HttpStatusCode.OK)
-            {
-                var resultJsonContent = JsonConvert.SerializeObject(result);
-                var resultUtf8Bytes = Encoding.UTF8.GetBytes(resultJsonContent);
+            this.Response.ContentLength = resultUtf8Bytes.Length;
+            this.Response.ContentType = SupportedContentType;
 
-                this.Response.ContentLength = resultUtf8Bytes.Length;
-                this.Response.ContentType = SupportedContentType;
+            await this.Response.Body.WriteAsync(resultUtf8Bytes, 0, resultUtf8Bytes.Length);
+        }
 
-                await this.Response.Body.WriteAsync(resultUtf8Bytes, 0, resultUtf8Bytes.Length);
-            }
+        async Task SendErrorResponse(HttpStatusCode status, string errorMsg)
```

Comment:
Should this be consistent to return error in json format?

Reply:
Split EdgeHubScopeResult into success/error types, and refactored the result construction to match.  Now that the response is always a EdgeHubScopeResult we can have errors in Json format.